### ISSUE TITLE
Confusion around when to use timeout vs recv_timeout

### ISSUE THREAD

**Issue by Cohen-Carlisle** [edgurgel/httpoison#215]
Getting `	{:error, %HTTPoison.Error{id: nil, reason: :timeout}}` in about 6 seconds with my options set as `[timeout: 30_000]`. After much digging and confusion, it seems that the timeout is stemming from `recv_timeout` being too low. I found this confusing for 2 reasons:
1) `reason: :timeout` is reported, which suggests that the `timeout` is the appropriate option to adjust.
2) The docs around `timeout` vs `recv_timeout` are pretty vague... I'm still not sure exactly what each one does and how they are different.

**Comment by edgurgel** [comment]
Yeah the struggle is real: https://github.com/edgurgel/httpoison/issues/211

The only way to solve this issue is to patch hackney so that it will provide the correct timeout error.

**Comment by sschneider1207** [comment]
@edgurgel hackney returns `{:error, :connect_timeout}` when the `connect_timeout` is reached, and `{:error, :timeout}` when the `recv_timeout` is reached.

**Comment by Cohen-Carlisle** [comment]
Any plans to do anything further with this? I think the new(-ish) docs around setting `:timeout` and `:recv_timeout` in request options are great, but there is still a lot of potential for confusion around getting back `:timeout | :connect_timeout` error reasons and trying to set the corresponding request options. I see a few approaches:
1. Change the request options to match the error reasons, or vice versa. Unfortunately, it would be backwards incompatible.
2. Introduce a `:connect_timeout` option that is an alias of `:timeout`. Doesn't solve the problem of getting a `:timeout` reason and setting the (incorrect) `:timeout` option.
3. Add more docs around the specific timeout reasons and which request option to set. This would be helpful no matter what else was done.
I might be able to carve out some time to do number 3.
